### ISSUE TITLE
Conflict with Servo Library

### ISSUE THREAD

**Issue by ageppert** [TMRh20/TMRpcm#3]
I'm not sure how to get ahold of you, so I hope this is OK. I am using your audio player, and it works great. But I'd like to also use the servo library associated with another timer. I think the audio library is conflicting with the servo library. I am using an Arduino Uno. Both libraries work fine when run in separate programs. Pin assignments:
# define servo1_pin         3  // Output  (Uno Timer2 PWM pin)
# define speaker_output     9  // Output  (Uno Timer1 PWM pin)

Any idea why these might be conflicting? I have tried all of the other PWM pins for the servo, and it's dead on all of them, except it jitters on pin 10 (Uno Timer1) which is associated with the audio output pin Timer1.
This is for a Valentine's Day project I'm helping my son with, due Friday as you might guess.
Thanks in advance for you assistance. You can see the project code here:
https://github.com/ageppert/Valentine_Robot

-Andy

**Comment by piing123** [comment]
Hi, i have the same problem asageppert. For my project, i wanted to play a wav file after a force sensor detected a force and then move a servo motor. I have tried the ServoTimer2 library to avoid the conflict between the servo and tmrpcm libraries where they use timer 1 as default. But, the wav file is able to play the first time and the servo is able to move. But, after the first time, only the servo motor is able to move without playing the wav file. The below is my code. Hope anyone can solve my problem. Tyvm.


#include"ServoTimer2.h"
#include <SD.h>                      // need to include the SD library
#define SD_ChipSelectPin 53  //example uses hardware SS pin 53 on Mega2560
//#define SD_ChipSelectPin 4  //using digital pin 4 on arduino nano 328, can use other pins
#include <TMRpcm.h>           //  also need to include this library...
#include <SPI.h>

TMRpcm tmrpcm;   // create an object for use in this sketch
ServoTimer2 myServo1;

void setup(void) {
  tmrpcm.speakerPin = 6; //5,6,11 or 46 on Mega, 9 on Uno, Nano, etc
  tmrpcm.setVolume(5);
  Serial.begin(9600);  
  if (!SD.begin(SD_ChipSelectPin)) {  // see if the card is present and can be initialized:
    Serial.println("SD fail");  
    return;   // don't do anything more if not
  }
  
 fsrReading1 = analogRead(fsrPin1);       //force sensor to take reading
 Serial.print("Analog reading1 = ");
 Serial.println(fsrReading1);   

void loop(void) 
{
  while (true)                       // FSR 1   (head)
  {
    if ( fsrReading1 <= 10) 
    {  
      Serial.println("nothing"); 
      break;
    } 
    else if (fsrReading1 < 200) 
    {
      Serial.println("neutral/sad");    // neutral or sad emotion
      tmrpcm.play("sad.wav");
      delay(4000);
      left_go();           //(left hand move)
      left_go();           //(left hand move back)
      break;
     }
    else
     {
      Serial.println("nothing"); 
      break;
     }
  }
}
void left_go()
{
  delay(1000);
  myServo1.attach(5);
  for (int i =750; i <= 1250; i++)    // degree (0-60)
  {
    myServo1.write(i);
    delay(3);
  }
  delay(500);
  myServo1.detach();
  delay(1000);
}
void left_return()
{
  delay(1000);
  myServo1.attach(5);
  for (int i =1250; i >= 750; i--)    //degree (60-0)
  {
    myServo1.write(i);
    delay(3);
  }
  delay(500);
  myServo1.detach();
  delay(1000);
}
